lib/internal/readline/interface.js

```diff
@@ -1058,31 +1070,58 @@ class Interface extends InterfaceConstructor {
     const col = this.columns;
     let rows = 0;
     str = stripVTControlCharacters(str);
-    for (const char of new SafeStringIterator(str)) {
-      if (char === '\n') {
-        // Rows must be incremented by 1 even if offset = 0 or col = +Infinity.
-        rows += MathCeil(offset / col) || 1;
-        offset = 0;
-        continue;
+
+    // Split the string into lines for better handling of multiline scenarios
+    const lines = StringPrototypeSplit(str, '\n');
```

Comment:
Note: I think this is fine for the REPL while it's always faster to directly iterate over the string instead. Worst case is pretty much an outer loop with an if where you slice the part out of the string. That way there is no additional memory needed besides a few pointers and the state handling.